io/src/lib.rs

```diff
@@ -13,9 +13,9 @@
 //! This traits are not one-for-one drop-ins, but are as close as possible while still implementing
 //! `std::io`'s traits without unnecessary complexity.
 
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
FTR I'm used to unconditionally setting `no_std` and conditionally `extern crate std`. I'm not sure what all pros and cons are but at least I find `#![no_std]` more readable than this thing.